### DISCUSSION TITLE
Remix AWS using API Gateway

### DISCUSSION THREAD

**Discussion by hakjeri** [remix-run/remix#5109] [Q&A]
I am having trouble deploying remix on AWS with a custom domain, following a mix of the guides found on https://github.com/remix-run/grunge-stack and https://github.com/remix-run/starter-architect.

The issue is that I get everything to deploy on AWS API Gateway and woriking for TLS 1.2 traffic, but it does not handle unencrypted calls at all. Ideally, I'd like to redirect http to https, but API Gateway does not seem to handle this. And my attempts at using ALB/ELB or CloudFront has not been working.

Are there any guides or tips to handling http traffic for using AWS API Gateway and remix?

**Comment by HerrBertling** [comment]
Sounds more like an AWS issue than a Remix thing – apparently, API Gateway does not support calls to `http://…`? I found this post that might help you – you'd have to set up CloudFront to redirect the http traffic to https. https://blog.hyper.io/using-cloudfront-to-redirect-http-to-https-on-apigateway/

**Reply by hakjeri** [reply]
Yes that would have been a solution, but after some more digging it seems that Cloudfront requires that the TLS certificate to be located in us-east-1, whereas I would prefer to keep all my resources in eu-north-1 or at least somewhere in Europe.
